### PR TITLE
nixos: Add missing packages

### PR DESCRIPTION
**Pull request by kjeremy**
This adds a lot of missing packages for nixos. Verified with https://search.nixos.org/packages and cross referenced with https://packages.ubuntu.com/

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -7371,6 +7400,7 @@ opencascade:
     stretch: [liboce-visualization-dev]
   fedora: [opencascade-devel]
   gentoo: [sci-libs/opencascade]
+  nixos: [opencascade]
```
Comment (MichaelOrlov) [inline rosdep/base.yaml:7403]: @kjeremy I was not able to find `opencascade` in the nixos index.
https://search.nixos.org/packages?channel=24.05&from=0&size=50&sort=relevance&type=packages&query=opencascade

File: rosdep/base.yaml [hunk 2]
```diff
@@ -7652,20 +7684,23 @@ postgresql-client:
...
   gentoo: [dev-db/postgresql]
+  nixos: [postgresql]
   ubuntu: [postgresql-client]
 postgresql-postgis:
   debian:
     buster: [postgresql-11-postgis-2.5, postgresql-contrib, postgresql-server-dev-all]
     stretch: [postgresql-9.6-postgis-2.3, postgresql-contrib, postgresql-server-dev-all]
   fedora: [postgis]
   gentoo: [dev-db/postgis, =dev-db/postgresql-9*]
+  nixos: [postgresqlPackages.postgis]
```
Comment (MichaelOrlov) [inline rosdep/base.yaml:7695]: There is no `postgresqlPackages.postgis` in the nixos index.
However, there present pinned versions like [postgresql16Packages.postgis](https://search.nixos.org/packages?channel=24.05&from=0&size=50&sort=relevance&type=packages&query=postgis) or [postgresql15Packages.postgis](https://search.nixos.org/packages?channel=24.05&from=0&size=50&sort=relevance&type=packages&query=postgis)

I think we need to refer to one of the pinned versions.

**Review by MichaelOrlov — CHANGES_REQUESTED** [review]
I've verified all referenced packages in the nixos index https://search.nixos.org/packages.

I was not able to find a direct match for 2 of them. See my comments inilne.

File: rosdep/base.yaml (hunk 2 above)
Comment (kjeremy) [reply rosdep/base.yaml:7695]: It's an alias like `python3Packages`:


```console
[jkolb@nixos:~/nix-ros-overlay]$ nix build nixpkgs#postgresqlPackages.postgis

[jkolb@nixos:~/nix-ros-overlay]$ ls -l result
lrwxrwxrwx 1 jkolb users 57 Oct  3 09:37 result -> /nix/store/366llrbbdjw0dim6xh4m01dkncg0mmv1-postgis-3.4.3
```

**Comment by kjeremy** [comment]
@MichaelOrlov thanks for the review. Everything should be all set now.

**Review by clalancette — APPROVED** [review]

File: rosdep/base.yaml (hunk 2 above)
Comment (MichaelOrlov) [reply rosdep/base.yaml:7695]: Ok, then. Thanks for the clarification.

**Review by MichaelOrlov — APPROVED** [review]
Verified. LGTM.

**Pull request merged**